tests/e2e/specs/miscellaneous/CheckUserEnvsTest.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { expect } from 'chai';
+import { Dashboard } from '../../pageobjects/dashboard/Dashboard';
+import { BrowserTabsUtil } from '../../utils/BrowserTabsUtil';
+import { OAUTH_CONSTANTS } from '../../constants/OAUTH_CONSTANTS';
+
+suite('"User secrets in workspace" test', function (): void {
+	const workspaceHandlingTests: WorkspaceHandlingTests = e2eContainer.get(CLASSES.WorkspaceHandlingTests);
+	const projectAndFileTests: ProjectAndFileTests = e2eContainer.get(CLASSES.ProjectAndFileTests);
+	const loginTests: LoginTests = e2eContainer.get(CLASSES.LoginTests);
+	const dashboard: Dashboard = e2eContainer.get(CLASSES.Dashboard);
+	const browserTabsUtil: BrowserTabsUtil = e2eContainer.get(CLASSES.BrowserTabsUtil);
+
+	let kubernetesCommandLineToolsExecutor: KubernetesCommandLineToolsExecutor = e2eContainer.get(
+		CLASSES.KubernetesCommandLineToolsExecutor
+	);
+	const shellExecutor: ShellExecutor = e2eContainer.get(CLASSES.ShellExecutor);
+	const testWorkspaceUtil: ITestWorkspaceUtil = e2eContainer.get(TYPES.WorkspaceUtil);
+	const userProject: string = OAUTH_CONSTANTS.TS_SELENIUM_OCP_USERNAME + `-devspaces`;
+	const stackName: string = 'Empty Workspace';
+
+	const mountEnvCommand: string = `echo '
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mount
+  labels:
+    controller.devfile.io/mount-to-devworkspace: "true"
+    controller.devfile.io/watch-secret: "true"
+  annotations:
+    controller.devfile.io/mount-as: env
+data:
+  mykey: bXl2YWx1ZQ==
+type: Opaque
+' | oc apply -f -`;
+
+	suiteSetup('Mount kubernetes secret to the DevWorkspace containers', function (): void {
+		kubernetesCommandLineToolsExecutor.loginToOcp();
+		shellExecutor.executeCommand('oc project ' + userProject);
+		shellExecutor.executeCommand('oc delete secret mount || true');
+
+		ShellString(shellExecutor.executeCommand(mountEnvCommand));
+	});
+
+	suiteSetup('Login', async function (): Promise<void> {
+		await loginTests.loginIntoChe();
+	});
+
+	test(`Create and open new workspace, stack:${stackName} and wait workspace readiness`, async function (): Promise<void> {
+		await workspaceHandlingTests.createAndOpenWorkspace(stackName);
+		await workspaceHandlingTests.obtainWorkspaceNameFromStartingPage();
+
+		registerRunningWorkspace(WorkspaceHandlingTests.getWorkspaceName());
+		await projectAndFileTests.waitWorkspaceReadinessForCheCodeEditor();
+	});
+
+	test('Check that mounted secret exists in envs list', async function (): Promise<void> {
+		kubernetesCommandLineToolsExecutor = e2eContainer.get(CLASSES.KubernetesCommandLineToolsExecutor);
+		kubernetesCommandLineToolsExecutor.workspaceName = WorkspaceHandlingTests.getWorkspaceName();
+		kubernetesCommandLineToolsExecutor.loginToOcp();
+		kubernetesCommandLineToolsExecutor.getPodAndContainerNames();
+
+		const output: ShellString = kubernetesCommandLineToolsExecutor.execInContainerCommand('env | grep mykey');
+		expect(output, 'User env not found').contains('mykey=myvalue');
```

Comment:
IMHO, it will make the test result more clear if we log the following error message: 
> Environment variable "mykey" from the user secret is not found